homeassistant/components/sensibo/switch.py

```diff
@@ -178,3 +165,22 @@ def extra_state_attributes(self) -> Mapping[str, Any] | None:
         if self.entity_description.extra_fn:
             return self.entity_description.extra_fn(self.device_data)
         return None
+
+    @api_call_decorator
+    async def api_call(
+        self, device_data: SensiboDevice, key: Any, value: Any, command: str
+    ) -> bool:
+        """Make service call to api."""
+        result = {}
+        data = build_params(command, self.device_data)
```

Comment:
Why would this ever return None?  Should it raise a ValueError instead?

Reply:
command `del_timer` doesn't have any params and therefore returning None but given you feedback I think it should raise in case it's wrong somehow so I explicit set `del_timer` to return None so any other would raise `ValueError`

Reply:
I think you could get rid of the if TYPE_CHECKING and call build_params only in the places its needed

Reply:
Looking at this some more the `api_call` and `build_params` end up defeating the value of typing to make sure we get things right. 

It might make more sense to create subclasses with specific async_turn_on/async_turn_off so we know we get the typing right

Reply:
I splitted out the functions now as suggested. Probably also easier to see what's going on..